owa/app/js/components/manage_patient_identifier_sources/edit_patient_identifier_sources/localPoolIdentifiers.jsx

```diff
@@ -0,0 +1,170 @@
+      sources: []
+    }
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.createSelectOptions = this.createSelectOptions.bind(this);
+    this.handleSelectChange = this.handleSelectChange.bind(this);
+  }
+
+  componentWillMount(){
+    this.createSelectOptions();
+  }
+
+  createSelectOptions(){
+    apiCall(null, 'get', '/idgen/identifiersource?identifierType=' + this.state.identifierTypeUuid).then((response) => {
+      let sources = response.results;
+      let currentIndex = 0;
+      Object.keys(sources).map(index => {
+        if(sources[index].uuid === this.state.uuid){
+          currentIndex = index;
+        }
+      });
+      delete sources[currentIndex];
+      this.setState({sources: response.results});
+    });
+  }
+
+  handleChange(event){
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  }
+
+  handleSelectChange(){
+    this.setState({
+      sourceUuid: document.getElementById("sourceUuid").value
+    });
+  }
+
+  handleRadioChange(name) {
+    if(name === "sequential"){
+      this.setState({
+        sequential: !this.state.sequential
+      });
+    }
+    if(name === "refill"){
+      this.setState({
+        refillWithScheduledTask: !this.state.refillWithScheduledTask
+      });
+    }
+  }
+
+  handleSubmit(){
+    let data = {
+      name: this.state.name,
+      description: this.state.description,
+      sourceUuid: this.state.sourceUuid,
+      sequential: this.state.sequential,
+      refillWithScheduledTask: this.state.refillWithScheduledTask,
+      batchSize: this.state.batchSize,
+      minPoolSize: this.state.minPoolSize
+    };
+
+    apiCall(data, 'post', '/idgen/identifiersource/' + this.state.uuid).then((response) => {
+      if( response["error"] === undefined ) {
+        this.props.handleUpdateName(this.state.index, this.state.name);
+        this.props.handleAlerts("success", "Successfull Updated " + this.state.name); 
+      }else{
+        this.props.handleAlerts("error", response["error"]["message"]);
+      }
+    });
+  }
+
+  render() {
+
+    return (
+      <div>
+        <div className="editPatientIdentifierSourcesTitle">Edit: Local Pool of Identifiers for {this.props.identifierSource.identifierTypeName}</div>
+        <Table size="sm" responsive>
+          <thead>
+            <tr>
+              <td className="editPatientIdentifierSourcesTitle">Field</td>
+              <td className="editPatientIdentifierSourcesTitle">Value</td>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td className="editPatientIdentifierSourcesTitle"><span className="requiredField">*</span> Name:</td>
+              <td>
+                <Input id="name" value={this.state.name == null ? "" : this.state.name} onChange={event => this.handleChange(event)} name="name" className="editPatientIdentifierSourcesInput" size="sm" type="text" />
```

Comment:
You should use `===` instead of `==`. Also you can reduce the line by doing `this.state.name ? this.state.name : ''`. It will always return `''` if this.state.name is equals to `null, undefined or an empty string`

Reply:
fixed